From 4a63a848c062851776123d4bc7ec10eb498a70dd Mon Sep 17 00:00:00 2001
From: lijuang <lijuang@codeaurora.org>
Date: Thu, 12 Oct 2017 10:54:50 +0800
Subject: QcomModulePkg: Correct the array initialization for VerifiedBootMenu

The array size of ID and StartlineSpace are 4, but they are initialized
with a string which its length is 4.  '\0' is needed for each string array

Change-Id: I59224ccb2868635b009289158db5505d74de4ab0
---
 QcomModulePkg/Library/BootLib/VerifiedBootMenu.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/QcomModulePkg/Library/BootLib/VerifiedBootMenu.c b/QcomModulePkg/Library/BootLib/VerifiedBootMenu.c
index a8810c7..3d17e7e 100644
--- a/QcomModulePkg/Library/BootLib/VerifiedBootMenu.c
+++ b/QcomModulePkg/Library/BootLib/VerifiedBootMenu.c
@@ -187,8 +187,8 @@ STATIC VOID GetDisplayOutPut(CHAR8 *Target, UINTN TargetLen, CHAR8 *Source,
 	/* First line starts with 4 characters of "ID: ",
 	   other lines start with 4 spaces to make the length of each line 21
 	*/
-	CONST CHAR8 ID[4] = "ID: ";
-	CONST CHAR8 StartlineSpace[4] = "    ";
+    CONST CHAR8 Id[] = "ID: ";
+    CONST CHAR8 StartlineSpace[] = "    ";
 
 	/* Each line contains 21 characters (4 spaces in the beginning),
 	16 characters from fingerprint, one character for endline */
@@ -197,7 +197,7 @@ STATIC VOID GetDisplayOutPut(CHAR8 *Target, UINTN TargetLen, CHAR8 *Source,
 
 	/* each line contains 4 spaces at the beginning and one endline
 	   character at the end */
-	FinalLen = (sizeof(ID) + 1) * (NumberOfLines) + SourceLen;
+    FinalLen = (AsciiStrLen (Id) + 1) * (NumberOfLines) + SourceLen;
 
 	/* if final size is bigger that display output size,
 	   reduce the numbe of lines,
@@ -210,11 +210,13 @@ STATIC VOID GetDisplayOutPut(CHAR8 *Target, UINTN TargetLen, CHAR8 *Source,
 
 	for (LineNum= 0; LineNum < NumberOfLines; LineNum++) {
 		if (LineNum == 0) {
-			gBS->CopyMem(Target + TargetIndex, (VOID *)ID, sizeof(ID));
+            gBS->CopyMem (Target + TargetIndex, (VOID *)Id,
+                        AsciiStrLen (Id));
 		} else {
-			gBS->CopyMem(Target + TargetIndex, (VOID *)StartlineSpace, sizeof(StartlineSpace));
+            gBS->CopyMem (Target + TargetIndex, (VOID *)StartlineSpace,
+                        AsciiStrLen (StartlineSpace));
 		}
-		TargetIndex = TargetIndex + sizeof(ID);
+        TargetIndex = TargetIndex + AsciiStrLen (Id);
 		if ((SourceLen - SourceIndex) >= FINGERPRINT_LINE_LEN) {
 			gBS->CopyMem(Target + TargetIndex,
 				Source + SourceIndex, FINGERPRINT_LINE_LEN);
-- 
cgit v1.1

